packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe(
+        'onCreate',
+        [
+          async () => {
+            await user1TodoHelper.create(createResultSetName, todoRandom, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, privateContent: null, ownerContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe(
+        'onUpdate',
+        [
+          async () => {
+            await user1TodoHelper.update(updateResultSetName, todoRandomUpdated, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onUpdateSubscriptionResult[0],
+        { ...todoRandomUpdated, privateContent: null, ownerContent: null },
+        `onUpdate${modelName}`,
+      );
+
+      /*
+      const onDeleteSubscriptionResult = await subTodoHelper.subscribe(
+        'onDelete', 
+        [
+          async () => {
+            await user1TodoHelper.delete(deleteResultSetName, { customId: todoRandom.customId }, completeOwnerResultSet);
+          },
+        ], 
+        {}, 
+        completeOwnerResultSet,
+        false
+      );
+      expect(onDeleteSubscriptionResult).toHaveLength(1);
+      checkOperationResult(onDeleteSubscriptionResult[0], {...todoRandomUpdated, privateContent: null, ownerContent: null}, `onDelete${modelName}`);
+      */
+    });
+
+    test('Custom owner model auth and restricted field operations', async () => {
+      const modelName = 'TodoCustomOwnerContentVarious';
+      const user1ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName1], schema);
+      const user2ModelOperationHelpers = createModelOperationHelpers(appSyncClients[userPoolProvider][userName2], schema);
+      const user1TodoHelper = user1ModelOperationHelpers[modelName];
+      const user2TodoHelper = user2ModelOperationHelpers[modelName];
+
+      const todo = {
+        customId: Date.now().toString(),
+        author: userName1,
+        privateContent: 'Private Content',
+        ownerContent: 'Owner Content',
+      };
+      const createResultSetName = `create${modelName}`;
+      const updateResultSetName = `update${modelName}`;
+      const deleteResultSetName = `delete${modelName}`;
+      const privateResultSet = `
+        customId
+        privateContent
+      `;
+      const ownerResultSet = `
+        ${privateResultSet}
+        author
+      `;
+      const completeOwnerResultSet = `
+        ${ownerResultSet}
+        ownerContent
+      `;
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket